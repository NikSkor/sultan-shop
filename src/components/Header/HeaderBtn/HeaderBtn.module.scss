@import '../../../vars.scss';

.btn {
  background-color: #FFC85E;
  border-radius: 79px;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 21px 22px;
  gap: 12px;
  font-weight: 700;
  font-size: 14px;
  line-height: 17px;
  color: #FFFFFF;
  flex-wrap: nowrap;
  min-width: 192px;
  cursor: pointer;

  
}

.btnCatalog {
  @include tablet {
    background-color: transparent;
    color: #3F4E65;
    flex-direction: row-reverse;
    align-items: center;
    grid-area: catalog;
    min-width: none;
  }
}

.price{
  @include laptop {
      display: none;
    }
}

.text {
  @include tablet {
    font-style: normal;
    font-weight: 600;
    font-size: 12px;
    line-height: 15px;
  }
}