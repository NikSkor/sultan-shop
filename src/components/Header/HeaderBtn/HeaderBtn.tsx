import React, { FC } from 'react';
import style from './HeaderBtn.module.scss';

interface BtnProps {
  text: string,
  children: React.ReactNode,
  isPrice?: boolean,
  isHeader?: boolean
}

const HeaderBtn: FC<BtnProps> = ({text, children, isPrice, isHeader}) => {
  return (
    <button className={`${style.btn} ${isPrice === true ? style.price : ''} ${isHeader === true ? style.btnCatalog : ''}`}>
      <div className={style.text}>{text}</div>
      {children}
    </button>
  )
}

export default HeaderBtn;