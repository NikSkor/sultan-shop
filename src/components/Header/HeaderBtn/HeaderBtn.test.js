import HeaderBtn from './HeaderBtn';
import renderer from 'react-test-renderer';
import img from '../../../img/basket.svg';
import { cleanup } from '@testing-library/react';


describe('test of headerBtn component', () => {
afterEach(cleanup);

    test('component renders props correctly, truthy values', () => {
      const tree = renderer
        .create(
          <HeaderBtn text='Список' isPrice={false}>
            <img src={img} alt='Значок прайс листа' />
          </HeaderBtn>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('component renders props correctly, falsy values', () => {
      const tree = renderer.create(<HeaderBtn />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});
