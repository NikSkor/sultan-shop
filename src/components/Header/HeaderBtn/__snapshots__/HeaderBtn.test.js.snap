// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test of headerBtn component component renders props correctly, falsy values 1`] = `
<button
  className="btn  "
>
  <div
    className="text"
  />
</button>
`;

exports[`test of headerBtn component component renders props correctly, truthy values 1`] = `
<button
  className="btn  "
>
  <div
    className="text"
  >
    Список
  </div>
  <img
    alt="Значок прайс листа"
    src="basket.svg"
  />
</button>
`;
